src/api/cvc4cpp.cpp

```diff
@@ -3691,24 +3706,56 @@ Term Solver::mkPi() const
   CVC4_API_SOLVER_TRY_CATCH_END;
 }
 
+Term Solver::mkInteger(const std::string& s) const
+{
+  CVC4_API_SOLVER_TRY_CATCH_BEGIN;
+  Term integer = mkRealFromStrHelper(s);
+  CVC4_API_ARG_CHECK_EXPECTED(integer.getSort() == getIntegerSort(), s)
+      << " an integer";
+  return integer;
+  CVC4_API_SOLVER_TRY_CATCH_END;
+}
+
+Term Solver::mkInteger(int64_t val) const
+{
+  CVC4_API_SOLVER_TRY_CATCH_BEGIN;
+  Term integer = mkValHelper<CVC4::Rational>(CVC4::Rational(val));
+  Assert(integer.getSort() == getIntegerSort());
+  return integer;
+  CVC4_API_SOLVER_TRY_CATCH_END;
+}
+
 Term Solver::mkReal(const std::string& s) const
 {
   CVC4_API_SOLVER_TRY_CATCH_BEGIN;
-  return mkRealFromStrHelper(s);
+  Term rational = mkRealFromStrHelper(s);
+  return ensureRealSort(rational);
   CVC4_API_SOLVER_TRY_CATCH_END;
 }
 
+Term Solver::ensureRealSort(const Term t) const
+{
+  Assert(t.getSort() == getIntegerSort() || t.getSort() == getRealSort());
```

Comment:
This shouldn't be an exception unless `mkReal` guards and checks that the string passed in is indeed a Real and not an Integer sort. `mkRealFromStrHelper` only checks if it is a string that represents either an Int, Real or rational value.

Reply:
`ensureRealSort` is a private function. So for now, only `mkReal` functions call it. Still, if it does not affect the performance, I prefer to have this check to prevent passing terms with wrong sort.

Reply:
We discussed and resolved this offline.